zulipterminal/themes/color.py

```diff
@@ -0,0 +1,34 @@
+"""
+COLOR
+-----
+Contains color definitions or functions common across all themes.
+For further details on themefiles look at the theme contribution guide.
+"""
+from enum import Enum
+from typing import Type
+
+
+def color_properties(colors: Type[Enum], *prop: str) -> Type[Enum]:
+    """
+    Adds properties(Bold, Italics, etc...) to Enum Colors in theme files.
+    Useage: color_properties(Color, 'BOLD', 'ITALICS', 'STRIKETHROUGH')
+
+        NOTE: color_properties(Color, BOLD, ITALICS) would result in only
+        Color.WHITE and Color.WHITE__BOLD_ITALICS
+        but not Color.WHITE__BOLD or Color.WHITE__ITALICS.
+        One would also have to do color_properties(Color, BOLD)
+        and color_properties(Color, ITALICS) for the others to work
+        >>> This function can be later extended to achieve all combinations
+        with one call to the function.
+    """
+    prop_n = "_".join([p.upper() for p in prop])
+    prop_v = " , ".join([p.lower() for p in prop])
+    updated_colors: Type[Enum]
+    updated_colors = Enum(  # type: ignore
```

Comment:
I think these 2 are related to ignoring type hinting:
* https://github.com/python/mypy/issues/529
* https://github.com/python/mypy/issues/5317
I'll add them to the doc string as NOTE?

Reply:
The server repo has a style something like `foo = bar  # type: ignore # explanation` where the explanation can be a link if there's an open report for it. This might be too dynamic to type-check, essentially?

Generally enums don't seem to be type-checked very well using mypy :/
(eg. as of python/typing#535 `.value` is an `Any`)

Reply:
Re too dynamic, yes, I think whatever is inside the Enum functional call is too complicated for mypy to understand.